plugins/modules/cloud/scaleway/scaleway_database_backup.py

```diff
@@ -0,0 +1,346 @@
+    response = account_api.post(
+        '/rdb/v1/regions/%s/backups/%s/export' % (module.params.get('region'), backup['id']), {})
+
+    if response.ok:
+        result = wait_to_complete_state_transition(module, account_api, response.json)
+        module.exit_json(changed=True, data=result)
+
+    module.fail_json(msg='Error exporting backup [{0}: {1}]'.format(response.status_code, response.json))
+
+
+def restored_strategy(module, account_api, backup):
+    if backup is None:
+        module.fail_json(msg=('Backup "%s" not found' % module.params['id']))
+
+    database_name = module.params['database_name']
+    instance_id = module.params['instance_id']
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    backup = wait_to_complete_state_transition(module, account_api, backup)
+
+    payload = {'database_name': database_name, 'instance_id': instance_id}
+    response = account_api.post('/rdb/v1/regions/%s/backups/%s/restore' % (module.params.get('region'), backup['id']),
+                                payload)
+
+    if response.ok:
+        result = wait_to_complete_state_transition(module, account_api, response.json)
+        module.exit_json(changed=True, data=result)
+
+    module.fail_json(msg='Error restoring backup [{0}: {1}]'.format(response.status_code, response.json))
+
+
+state_strategy = {
+    'present': present_strategy,
+    'absent': absent_strategy,
+    'exported': exported_strategy,
+    'restored': restored_strategy,
+}
+
+
+def core(module):
+    state = module.params['state']
+    backup_id = module.params['id']
+
+    account_api = Scaleway(module)
+
+    if backup_id is None:
+        backup_by_id = None
+    else:
+        response = account_api.get('/rdb/v1/regions/%s/backups/%s' % (module.params.get('region'), backup_id))
+        status_code = response.status_code
+        backup_json = response.json
+        backup_by_id = None
+        if status_code == 404:
+            backup_by_id = None
+        elif response.ok:
+            backup_by_id = backup_json
+        else:
+            module.fail_json(msg='Error getting backup [{0}: {1}]'.format(status_code, response.json['message']))
+
+    state_strategy[state](module, account_api, backup_by_id)
+
+
+def main():
+    argument_spec = scaleway_argument_spec()
+    argument_spec.update(dict(
+        state=dict(default='present', choices=['absent', 'present', 'exported', 'restored']),
+        region=dict(required=True, choices=SCALEWAY_REGIONS),
+        id=dict(),
+        name=dict(required=False),
+        database_name=dict(required=False),
+        instance_id=dict(required=False),
+        expires_at=dict(),
+        wait=dict(type='bool', default=False),
+        wait_timeout=dict(type='int', default=300),
+        wait_sleep_time=dict(type='int', default=3),
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
```

Comment:
```suggestion
        supports_check_mode=True,
    )
```
you could restrict possible parameter combinations by using something from https://docs.ansible.com/ansible/latest/reference_appendices/module_utils.html (e.g. `required_if`, `required_together`, `mutually_exclusive`, etc.)
if certain parameters are ignored in some cases, it also makes sense to point it out in the doc, e.g.
```
- Ignored when I(state=absent).
```

Reply:
I used `required_if` and `required_together` (thanks for the tip !)